python/GafferUI/Widget.py

```diff
@@ -316,12 +316,20 @@ def setHighlighted( self, highlighted ) :
 		if highlighted == self.getHighlighted() :
 			return
 
-		self._qtWidget().setProperty( "gafferHighlighted", GafferUI._Variant.toVariant( highlighted ) )
+		widget = self._qtWidget()
+		if type( widget ) == QtWidgets.QWidget :
+			widget = widget.layout().itemAt( 0 ).widget()
```

Comment:
OK, if the convention is that all widgets that call the base constructor with anything not derived from QWidget are responsible for overriding setHighlighted, then the issue is that GafferUI.Button fails to do this, and we don't need to change Widget.

This now irrelevant to this PR, since you're happy to only highlight _CompareImageWidget, which derives from Frame, which passes a QWidget derived QFrame to the constructor, so is unaffected by this.

Let me know if you want me to put together a separate PR with a fix for this, with the stylesheet change to allow buttons to be highlighted ... not sure whether there is another use for highlighting buttons ... I guess our API sort of implies it should work.

Reply:
> OK, if the convention is that all widgets that call the base constructor with anything not derived from QWidget are responsible for overriding setHighlighted, then the issue is that GafferUI.Button fails to do this, and we don't need to change Widget.

I don't understand this - `GafferUI.Button` passes `QPushButton` to `Widget.__init__`, so wouldn't hit the new code in Widget anyway. Button doesn't need to override `setHighlighted` - we just need a stylesheet for `QPushButton` with the `gafferHighlighted` property. It would be `_CompareModePlugValueWidget` that would have needed to override `setHighlighted`.

> Let me know if you want me to put together a separate PR with a fix for this, with the stylesheet change to allow buttons to be highlighted

I don't think it's urgent - let's just do it when we actually need it.